components/org.wso2.carbon.identity.api.server.notification.sender/org.wso2.carbon.identity.api.server.notification.sender.v2/src/main/java/org/wso2/carbon/identity/api/server/notification/sender/v2/core/NotificationSenderManagementService.java

```diff
@@ -0,0 +1,475 @@
+        } catch (NotificationSenderManagementException e) {
+            throw handleException(e);
+        }
+    }
+
+    /**
+     * Update email sender details by name.
+     *
+     * @param senderName               Email sender's name.
+     * @param emailSenderUpdateRequest Email sender's updated configurations.
+     * @return Updated email sender.
+     */
+    public EmailSender updateEmailSender(String senderName, EmailSenderUpdateRequest emailSenderUpdateRequest) {
+
+        EmailSenderDTO dto = buildEmailSenderDTO(senderName, emailSenderUpdateRequest);
+        try {
+            EmailSenderDTO emailSenderDTO = notificationSenderManagementService.updateEmailSender(dto);
+            return buildEmailSenderFromDTO(emailSenderDTO);
+        } catch (NotificationSenderManagementException e) {
+            throw handleException(e);
+        }
+    }
+
+    /**
+     * Update sms sender details by name.
+     *
+     * @param senderName             SMS sender' name.
+     * @param smsSenderUpdateRequest SMS sender's updated configurations.
+     * @return Updated SMS sender.
+     */
+    public SMSSender updateSMSSender(String senderName, SMSSenderUpdateRequest smsSenderUpdateRequest) {
+
+        SMSSenderDTO dto = buildSMSSenderDTO(senderName, smsSenderUpdateRequest);
+        try {
+            SMSSenderDTO smsSenderDTO = notificationSenderManagementService.updateSMSSender(dto);
+            return buildSMSSenderFromDTO(smsSenderDTO);
+        } catch (NotificationSenderManagementException e) {
+            throw handleException(e);
+        }
+    }
+
+    /**
+     * Update push sender details by name.
+     *
+     * @param senderName               Push sender's name.
+     * @param pushSenderUpdateRequest Push sender's updated configurations.
+     * @return Updated push sender.
+     */
+    public PushSender updatePushSender(String senderName, PushSenderUpdateRequest pushSenderUpdateRequest) {
+
+        PushSenderDTO dto = buildPushSenderDTO(senderName, pushSenderUpdateRequest);
+        try {
+            PushSenderDTO pushSenderDTO = NotificationSenderServiceHolder.getNotificationSenderManagementService()
+                    .updatePushSender(dto);
+            return buildPushSenderFromDTO(pushSenderDTO);
+        } catch (NotificationSenderManagementException e) {
+            throw handleException(e);
+        }
+    }
+
+    private EmailSenderDTO buildEmailSenderDTO(EmailSenderAdd emailSenderAdd) {
+
+        EmailSenderDTO dto = new EmailSenderDTO();
+        dto.setName(emailSenderAdd.getName());
+        dto.setFromAddress(emailSenderAdd.getFromAddress());
+        dto.setSmtpPort(emailSenderAdd.getSmtpPort());
+        dto.setAuthType(emailSenderAdd.getAuthType());
+        dto.setSmtpServerHost(emailSenderAdd.getSmtpServerHost());
+        List<Properties> properties = emailSenderAdd.getProperties();
+        properties.forEach((prop) -> dto.getProperties().put(prop.getKey(), prop.getValue()));
```

Comment:
Lets make this null safe.

Reply:
addressed with [68da487](https://github.com/wso2/identity-api-server/pull/865/commits/68da487d4e1ee4a75c62127a2c97c86da65bb63f)